File: test/fruit_picker/lib/orange.dart
import 'channel.dart';
import 'dart:async';

class Orange {
  Orange(double juiciness) {
    Channel.channel.invokeMethod<void>('Orange(double)',
        <String, dynamic>{'juiciness': juiciness, 'orangeHandle': handle});
  }

  final int handle = Channel.nextHandle++;

  Future<void> squeeze(double pressure) {
    return Channel.channel.invokeMethod<void>('Orange#squeeze',
        <String, dynamic>{'pressure': pressure, 'handle': handle});
  }
}

File: test/fruit_picker/android/src/main/java/dev/fruit/fruit_picker/fruitlibrary/fruits/Orange.java
package dev.fruit.fruit_picker.fruitlibrary.fruits;

public class Orange {
  public Orange(Double juiciness) {
    if (juiciness == null) throw new IllegalArgumentException();
  }

  public void squeeze(Double pressure) {
    if (pressure == null) throw new IllegalArgumentException();
  }
}
